app/assets/javascripts/initializers/initializeVideoPlayback.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * This script hunts for video tags and initializes the correct player
+ * depending on the platform:
+ * - web: jwplayer
+ * - iOS/Android: Native player
+ *
+ * Once jwplayer is initialized there's no follow up actions to be taken.
+ * Mobile Native players send back information into the DOM in order to
+ * interact and update the UI, therefore a MutationObserver is registered.
+ */
+
+ /* eslint no-use-before-define: 0 */
+ /* eslint no-param-reassign: 0 */
+ /* eslint no-useless-escape: 0 */
+ /* global jwplayer */
+
+function initializeVideoPlayback() {
+  var nativeBridgeMessage;
+  var currentTime = "0";
+
+  function getById(name) {
+    return document.getElementById(name);
+  }
+
+  function isNativeIOS() {
+    return (
+      navigator.userAgent === 'DEV-Native-ios' &&
+      window &&
+      window.webkit &&
+      window.webkit.messageHandlers &&
+      window.webkit.messageHandlers.video
+    );
+  }
+
+  function isNativeAndroid() {
+    return (
+      navigator.userAgent === 'DEV-Native-android' &&
+      typeof AndroidBridge !== 'undefined' &&
+      AndroidBridge !== null &&
+      AndroidBridge.videoMessage !== undefined
+    );
+  }
+
+  function getParameterByName(name, url) {
+    if (!url) url = window.location.href;
+    name = name.replace(/[\[\]]/g, "\\$&");
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+      results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  }
+
+  function timeToSeconds(hms) {
+    var a;
+    if (hms.length < 3) {
+      return hms
+    } else if (hms.length < 6) {
```

Comment:
**thought:** At one point we may want to look into making this function a bit more robust. `1:2:3` for example has a lenght of less than 6 but `+a[1]` would result in `NaN`. Disregard this comment if we have full control over the formatting of `hms` and this can never happen.

Reply:
I think we do have full control over this, mainly because I couldn't find anywhere we share a "timestamped URL of the article".

What I mean by this is that I believe this is an open ended "hidden feature", as in I can share a link [like this one starting at 1:40](https://dev.to/terminusdb/world-of-woqlcraft-ep-16-knowledge-graph-for-nlp-1jhc?t=1:40) and it will work, but we don't have any built-in share feature that will include the current time of the player.

I just sent in a tweak for better supporting this timestamp parameter but will definitely put some more safety nets in this method in the follow up 👍🏼